### PR TITLE
Fix expense command behaviour

### PR DESCRIPTION
**Pull request by liakify**
*Changelog:*
- Change expense command to not use updateFilteredPersonList to avoid affecting GUI
- Update command result output
- Update tests
- Update JsonAdaptedExpense to properly store the involved participants
- Updated expense command behaviour:
1. If only payer is specified and it is currently in an activity view context, everyone in the current activity is taken to be involved.
2. On the contrary, if expense is created outside activity view context and only payer is specified, then only one person (the payer) is actually added into the activity and the expense is taken to not involve anyone else besides the payer.
3. If duplicate names are specified, they will only be added into the expense once and warnings will be shown accordingly.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java [hunk 1]
```diff
@@ -130,7 +140,27 @@ public CommandResult execute(Model model) throws CommandException {
             model.addActivity(activity);
         }
 
-        return new CommandResult(String.format(MESSAGE_SUCCESS, persons.size(), successMessage.toString()));
+        return new CommandResult(String.format(MESSAGE_SUCCESS,
+                payingPerson.getName(), amount, description, successMessage.toString()));
```
Comment (Aulud) [inline src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: The arguments `payingPerson.getName()` and `amount` are flipped - currently a successful command shows the success message `Expense of Alex Yeoh by 25.00 successfully created`.

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java [hunk 2]
```diff
@@ -41,8 +41,8 @@
             + PREFIX_EXPENSE + "10.0 "
             + PREFIX_DESCRIPTION + "Bubble tea";
 
-    public static final String MESSAGE_SUCCESS = "%d expense(s) successfully created as follows:\n%s\n";
-    public static final String MESSAGE_EXPENSE = "%s paid %s.\n  Description: %s\n";
+    public static final String MESSAGE_SUCCESS =
+            "Expense of %s by %s successfully created.\n  Description: %s\n  People involved: %s";
```
Comment (Aulud) [inline src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Is the space for after the newline intentional?

**Review by Aulud — COMMENTED** [review]
Looks okay on inspection - however, it seems the list of people owing the payee for an expense is not saved in the JSON output. Mind checking it out @liakify?

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java (hunk 1 above)
Comment (liakify) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: ohshit

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java (hunk 2 above)
Comment (liakify) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Intended for it to be a bit of an indentation, going to change to tabs

Comment (Aulud) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Sounds good. We'll clean up the indentation in one pass later.

**Review by daekoon — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java [hunk 3]
```diff
@@ -130,7 +140,27 @@ public CommandResult execute(Model model) throws CommandException {
...
+    }
+
+    /**
+     * Searches for a {@code Person} object from a given list of people using a search string.
+     * @param str The search string
+     * @param searchScope The {@code Person} list to search from
+     * @return The search result as a {@code Person} object
+     * @throws CommandException if the search result is not unique
+     */
+    public Person searchPerson(String str, List<Person> searchScope) throws CommandException {
```
Comment (daekoon) [inline src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Is this method called outside the class? If not, it should be a private method

Comment (liakify) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: idk i'm thinking of moving this method elsewhere at some point, cause currently the name searching is all over the place

so this method would contain the `if (exact match) {return exact match} else {return unique keyword match}` logic and activity, invite, disinvite and expense all can have consistent search logic

Comment (daekoon) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Is `searchScope` always the addressbook level currently? If yes, I created a method `findPersonByName` on modelmanager that lets you do an exact matching by a String. You can merge master to your branch and use that method to clean up your code a bit.

This way, I think you can keep the method as private and no need to extract out the logic into some other helper class. Addressbook/ModelManager provides the API to find users by keywords/exact match, and its up to the other class to decide how they want to use it.

Comment (liakify) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: the issue is that the search scope is sometimes activity level, other times address book level so it gets kind of messy and i can't reuse directly

Comment (daekoon) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: Hmm that makes sense actually. But I think its ok to keep the method in there as a private method, I don't see any other logic in our app needing for such a specific search function (keyword/exact search in both AddressBook level and ActivityBook level). It'll just be a private helper function to keep the code clean.

**Review by Aulud — APPROVED** [review]
Looks good to go.

File: src/main/java/seedu/address/logic/commands/ExpenseCommand.java (hunk 3 above)
Comment (liakify) [reply src/main/java/seedu/address/logic/commands/ExpenseCommand.java]: adjusted

**Pull request merged**